### PR TITLE
remove filename from line number nodes

### PR DESCRIPTION
**Pull request by JeffBezanson**
@carnaval is this what you want?

### CODE REVIEW

File: src/cgutils.cpp [hunk 1]
```diff
@@ -921,6 +921,8 @@ JL_DLLEXPORT Type *julia_type_to_llvm(jl_value_t *jt, bool *isboxed)
         return PointerType::get(lt, 0);
     }
     if (jl_is_bitstype(jt)) {
+        if (jt == (jl_value_t*)jl_long_type)
+            return T_size;
```
Comment (vtjnash) [inline src/cgutils.cpp]: we could probably also just generally generically cache these in `jst->struct_decl`

**Comment by toivoh** [comment]
I use the filenames (and line numbers) in line number nodes in `Debug.jl` to keep track of source location. How would I do that going forward?

**Comment by JeffBezanson** [comment]
File names are available in `LambdaInfo`, and also (in theory, hopefully soon to be in practice) via backtraces.

**Comment by toivoh** [comment]
But `LambdaInfo` is for compiled methods, right? There will be no way for a macro to determine which source files went into creating the AST that was given to it?

**Comment by carnaval** [comment]
Ah. That may be a problem. The frontend could insert a `'(meta push_lambda ,linfo)` around blocks that are passed to macros (like we are trying to do for inline functions) ?
or maybe a `meta push/pop_file` because we don't have the linfo yet.

Or we could still have those filenames in there as before, just seemed weird to me to have the same information duplicated everywhere.

**Comment by JeffBezanson** [comment]
We could put `(meta file x.jl)` in surface ASTs I suppose. Macros are a difficult case though because they can expand to a mix of code from different locations.

**Comment by toivoh** [comment]
Also, different lines in functions can come from different files, after macro expansion. Would that information still be representable?

**Comment by carnaval** [comment]
yes I think you would need to have such a pair of node in every "toplevel" block, meaning the outer most block of a quote, including implicit quotes such as the argument to a macro. Feels a little better to me than having the file name everywhere but might not be worth the effort.

As an added benefit, those nodes could specify that the code is being quoted/macroed so we could display that in the backtrace maybe. Just thinking out loud.

**Comment by JeffBezanson** [comment]
Filename push/pop per outer `quote` block sounds pretty reasonable to me.

**Comment by toivoh** [comment]
Sure, that would be fine.

**Comment by JeffBezanson** [comment]
Hmm, adding "pop" at the end of blocks is actually tricky, since the last expression in a block determines its value. We might be able to handle this later in lowering, e.g. inserting "pop" only when merging one block into another in a non-final position.

**Pull request closed without merging**